Calvin/register frontend

claalmve opened the pull request:
This feature constructs a fully functional sign-in, sign-up, and sign-out process for different organizations.

WIP: Only sign-up is completely functional now, the others are still in development.

## Related PRs

_Optional - any related PRs you're waiting on, or PRs that will conflict, etc_

## Migrations

_Optional - if you added anything to the database through migration(s)_

## Screenshots

_Required for visual changes_

CC: @kwfk

claalmve committed:
feat(signup frontend): laying groundwork for signup frontend

claalmve committed:
feat(sign up): making signup flow work from front-end to back-end

claalmve committed:
Delete signup.tsx

claalmve committed:
Delete other-signup.tsx

kwfk commented:
The folder component structure should only be used with page-specific components. For actual pages, you want to use the regular file naming scheme with the stylesheet under `styles`

kwfk commented on pages/users/signup.module.css:
```diff
@@ -0,0 +1,32 @@
+  margin-top: 200px;
+  text-align: center;
+  margin-left: 35%;
+  margin-right: 30%;
```
strange styling?
you probably just use width 60%

kwfk replied:
and center with flexbox

kwfk commented on pages/users/signup/index.tsx:
```diff
@@ -0,0 +1,165 @@
+                    <Grid item xs={6}>
+                      <button className={styles.submit} type="submit">
+                        Create
+                      </button>
```
use a material UI button

kwfk commented on pages/users/signup/signup.module.css:
```diff
@@ -0,0 +1,57 @@
+.body {
+  font-family: Verdana;
+}
```
don't change font here

kwfk commented on pages/users/signup/signup.module.css:
```diff
@@ -0,0 +1,57 @@
+  height: 400px;
+  font-family: 'Montserrat', sans-serif;
+  font-weight: 800;
+}
```
weird styling?
use width 60% then center with flexbox and min-height
see #21 for example

kwfk replied:
also don't change font here

kwfk commented on pages/users/signup/index.tsx:
```diff
@@ -0,0 +1,165 @@
+                      &nbsp;
+                    </>
+                  ) : null}
+                  <Grid container xs={9} spacing={2}>
```
can't use xs on container

kwfk commented on pages/users/signup/index.tsx:
```diff
@@ -0,0 +1,165 @@
+
+  const handleSubmit = async (values: FormValues): Promise<any> => {
+    // Make post request
+    const res: any = await fetch('/api/users/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        email: values.email,
+        password: values.password,
+      }),
+    }).then((response) => {
+      if (response.ok) {
+        router.push('/');
+        return res;
+      }
+      setEmailExists(true);
+      return new Error('Email already exists.');
+    });
```
don't use chaining when using async await

kwfk reviewed (changes requested):
looking good! I think we can look a little more into the some of the types instead of using any

claalmve committed:
feat(merging in new main): merging in new main + adding signin

claalmve committed:
feat(finishing signup/signout): finishing signup/signout user flow

claalmve committed:
Merge branch 'calvin/register-frontend' of https://github.com/calblueprint/nbjc into calvin/register-frontend

kwfk committed:
refactor(signin and signup): refactored sign in and sign up pages to match other page components

Flattened sign in and sign up pages and moved styles to root styles directory. Also moved Join Us
button into condition in Layout component

kwfk commented on pages/users/signin.tsx:
```diff
@@ -0,0 +1,136 @@
+
+// UserSignIn page. Will need additional email verification to be able to create organizations.
+const UserSignIn: React.FC = () => {
+  const handleSubmit = async (values: FormValues): Promise<any> => {
```
return type should be `Promise<void>`

kwfk commented on pages/users/signin.tsx:
```diff
@@ -0,0 +1,136 @@
+      signIn('credentials', {
+        email: values.email,
+        password: values.password,
+        callbackUrl: 'http://localhost:3000',
```
change to `'/'`

kwfk commented on pages/users/signin.tsx:
```diff
@@ -0,0 +1,136 @@
+
+  const constructRow = (
+    varName: string,
+    handleChange: any,
```
handleChange type should be `FormikHandlers['handleChange']` imported from Formik

kwfk commented on pages/users/signin.tsx:
```diff
@@ -0,0 +1,136 @@
+    varName: string,
+    handleChange: any,
+    error?: string
+  ): any => (
```
return type should be `JSX.Element`

kwfk commented on pages/users/signup.tsx:
```diff
@@ -1,33 +1,180 @@
-    console.log(e.target.value);
+  const [emailExists, setEmailExists] = useState(false);
+
+  const handleSubmit = async (values: FormValues): Promise<any> => {
```
return type should be `Promise<void>`

kwfk commented on pages/users/signup.tsx:
```diff
@@ -1,33 +1,180 @@
+
+  const handleSubmit = async (values: FormValues): Promise<any> => {
+    // Make post request
+    const res: any = await fetch('/api/users/', {
```
`any` type here is not needed

kwfk commented on pages/users/signup.tsx:
```diff
@@ -1,33 +1,180 @@
+      signIn('credentials', {
+        email: values.email,
+        password: values.password,
+        callbackUrl: 'http://localhost:3000',
```
change to `'/'`

kwfk commented on pages/users/signup.tsx:
```diff
@@ -1,33 +1,180 @@
+  const constructRow = (
+    title: string,
+    varName: string,
+    handleChange: any,
```
handleChange type should be `FormikHandlers['handleChange']` imported from Formik

kwfk commented on pages/users/signup.tsx:
```diff
@@ -1,33 +1,180 @@
+    varName: string,
+    handleChange: any,
+    error?: string
+  ): any => (
```
return type should be `JSX.Element`

kwfk reviewed (changes requested):
nice work! I pushed a refactoring commit to flatten the structure of your page components so make sure to pull
some minor changes needed
the centering styling is still kinda wack

claalmve committed:
fix(signup/signin): fixing styling and error states

claalmve committed:
fix(signin): fixing signin logic

kwfk committed:
Merge branch 'main' into calvin/register-frontend

claalmve committed:
style(nextauth.ts): removing comments

kwfk committed:
feat(auth api): add custom error code and fix typo

kwfk committed:
feat(auth): fix styling and add isSubmitting progress

kwfk committed:
Merge branch 'calvin/register-frontend' of github.com:calblueprint/nbjc into calvin/register-frontend

kwfk committed:
fix(layout): return a loading state instead of nothing while session is loading

kwfk committed:
feat(auth): move signin and signup to be root level outside of /users

Moved signin and signup pages to be root level so that the route is just /signin or /signup. Also
moved Auth.module.css to root level of styles to match route. Added restricted page control with
session on signin and signup pages.

kwfk committed:
feat: add restricted page control with session

Restrict moderator and user settings pages so that only signed in users can access. Minor code style
change to Layout component.

kwfk committed:
Merge branch 'main' into calvin/register-frontend

kwfk reviewed (approved):
nice work XD

kwfk commented:
next step would be to look into how to change the user object in the session to have the fields that we want

kwfk committed:
fix(auth): update paths of signin and signup to reflect change to root

kwfk committed:
style(auth): change formik to use hooks instead of components

kwfk commented:
we will also want to look in to how to remove the previous page level errors

kwfk committed:
feat(signin signup): add type to formik
